filebeat/input/kafka/kafka_integration_test.go

```diff
@@ -126,6 +126,10 @@ func TestInput(t *testing.T) {
 		}
 	}
 
+	// sarama commits every second, we need to make sure
+	// all message acks are committed before the rest of the checks
+	<-time.After(2 * time.Second)
```

Comment:
I am not usually a fan of sleeps in tests as a synchronization mechanism. In this case it isn't an arbitrary number so I don't mind it.

Usually I prefer polling to observe the condition we are looking for. The `testify` package we already have as a dependency here has a nice `assert.Eventually` helper for this: https://pkg.go.dev/github.com/stretchr/testify/assert#Eventually

Reply:
The sarama client is incapsulated into the input implementation and is not accessible from here, additionally saram itself does not seem to expose the information when the client is `dirty` or it's synced. So, I don't see any other obvious options here how we could track the condition.